tkg/client/update_credentials.go

```diff
@@ -197,3 +212,120 @@ func (c *TkgClient) updateVSphereCredentialsForCluster(clusterClient clusterclie
 
 	return nil
 }
+
+// UpdateAzureClusterCredentials update azure cluster credentials
+func (c *TkgClient) UpdateAzureClusterCredentials(clusterClient clusterclient.Client, options *UpdateCredentialsOptions) error {
+	if options.AzureUpdateClusterOptions.AzureTenantID == "" || options.AzureUpdateClusterOptions.AzureClientID == "" || options.AzureUpdateClusterOptions.AzureClientSecret == "" {
+		return errors.New("either tenantId, clientId or clientSecret should not be empty")
+	}
+
+	if !options.IsRegionalCluster {
+		unifiedIdentity, err := clusterClient.CheckUnifiedAzureClusterIdentity(options.ClusterName, options.Namespace)
+		if err != nil {
+			return err
+		}
+		if unifiedIdentity {
+			log.Warningf("AzureCluster %s use the same AzureClusterIdentity with Management Cluster. It cannot be updated separately.", options.ClusterName)
+			return nil
+		}
+	}
+
+	if err := c.UpdateAzureCredentialsForCluster(clusterClient, options, true); err != nil {
+		return err
+	}
+
+	if options.IsRegionalCluster {
+		clusters, err := clusterClient.ListClusters("")
+		if err != nil {
+			return errors.Wrapf(err, "unable to list workload clusters")
+		}
+
+		for i := range clusters {
+			if clusters[i].Name == options.ClusterName {
+				continue
+			}
+
+			log.V(4).Infof("Updating credentials for workload cluster %q ...", clusters[i].Name)
+			err := c.UpdateAzureCredentialsForCluster(clusterClient, &UpdateCredentialsOptions{
+				ClusterName:               clusters[i].Name,
+				Namespace:                 clusters[i].Namespace,
+				IsRegionalCluster:         false,
+				IsCascading:               options.IsCascading,
+				AzureUpdateClusterOptions: options.AzureUpdateClusterOptions,
+			}, false)
+			if err != nil {
+				log.Error(err, "unable to update credentials for workload cluster")
+				continue
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *TkgClient) UpdateAzureCredentialsForCluster(clusterClient clusterclient.Client, options *UpdateCredentialsOptions, force bool) error {
```

Comment:
Please reorg the codes as:
1. mc will update the the wl clusters who has same azureidentity; if cascading, mc will update all wl clusters.
2. wl clusters will only come to here when it has its own azureidentity, so always update.

Reply:
The function is always called by single workload cluster.